packages/react-components/react-dialog/src/stories/DialogChangeFocus.stories.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { Dialog, DialogTrigger, DialogSurface, DialogTitle, DialogBody, DialogActions } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
+
+export const ChangeFocus = () => {
+  const buttonRef = React.useRef<HTMLButtonElement>(null);
+  const [open, setOpen] = React.useState(false);
+  React.useEffect(() => {
+    if (open && buttonRef.current) {
+      buttonRef.current.focus();
+    }
+  }, [open]);
+  return (
+    <>
+      <Dialog open={open} onOpenChange={(event, data) => setOpen(data.open)}>
+        <DialogTrigger>
+          <Button>Open dialog</Button>
+        </DialogTrigger>
+        <DialogSurface aria-label="label">
+          <DialogTitle>Dialog title</DialogTitle>
+          <DialogBody>This dialog focus on the second button instead of the first</DialogBody>
+          <DialogActions position="start">
+            <Button appearance="outline">Third Action</Button>
+          </DialogActions>
+          <DialogActions position="end">
+            <DialogTrigger>
+              <Button ref={buttonRef} appearance="secondary">
+                Close
+              </Button>
+            </DialogTrigger>
+            <Button appearance="primary">Do Something</Button>
+          </DialogActions>
+        </DialogSurface>
+      </Dialog>
+    </>
+  );
+};
+
+ChangeFocus.parameters = {
+  docs: {
+    description: {
```

Comment:
nit: Would be worth adding a description here, just saying that changing the default focused element can be done in an effect